yt_dlp/extractor/goodgame.py

```diff
@@ -1,40 +1,48 @@
 from .common import InfoExtractor
 from ..utils import (
-    clean_html,
     int_or_none,
     str_or_none,
     traverse_obj,
+    url_or_none,
 )
 
 
 class GoodGameIE(InfoExtractor):
     IE_NAME = 'goodgame:stream'
-    _VALID_URL = r'https?://goodgame\.ru/channel/(?P<id>\w+)'
+    _VALID_URL = r'https?://goodgame\.ru/(?:channel/)?(?P<id>[\w.*-]+)'
     _TESTS = [{
-        'url': 'https://goodgame.ru/channel/Pomi/#autoplay',
+        'url': 'https://goodgame.ru/TGW#autoplay',
         'info_dict': {
-            'id': 'pomi',
+            'id': '7998',
             'ext': 'mp4',
-            'title': r're:Reynor vs Special \(1/2,bo3\) Wardi Spring EU \- playoff \(финальный день\) \d{4}-\d{2}-\d{2} \d{2}:\d{2}$',
-            'channel_id': '1644',
-            'channel': 'Pomi',
-            'channel_url': 'https://goodgame.ru/channel/Pomi/',
-            'description': 'md5:4a87b775ee7b2b57bdccebe285bbe171',
-            'thumbnail': r're:^https?://.*\.jpg$',
+            'channel_id': '7998',
+            'title': r're:шоуматч Happy \(NE\) vs Fortitude \(UD\), потом ладдер и дс \d{4}-\d{2}-\d{2} \d{2}:\d{2}$',
+            'channel_url': 'https://goodgame.ru/TGW',
+            'thumbnail': 'https://hls.goodgame.ru/previews/7998_240.jpg',
+            'uploader': 'TGW',
+            'channel': 'JosephStalin',
             'live_status': 'is_live',
-            'view_count': int,
+            'age_limit': 18,
+            'channel_follower_count': int,
+            'uploader_id': '2899',
+            'concurrent_view_count': int,
         },
         'params': {'skip_download': 'm3u8'},
-        'skip': 'May not be online',
+    }, {
+        'url': 'https://goodgame.ru/Mr.Gray',
```

Comment:
can we also have one of these in the matching tests
```suggestion
        'url': 'https://goodgame.ru/channel/Mr.Gray',
```